plugins/inventory/proxmox.py

```diff
@@ -263,12 +264,16 @@ def _get_members_per_pool(self, pool):
 
     def _get_node_ip(self, node):
         ret = self._get_json("%s/api2/json/nodes/%s/network" % (self.proxmox_url, node))
+        conn_addr = socket.gethostbyname(re.findall(r'://([\w\-\.]+)', self.proxmox_url)[0])
 
+        # Try to verify the connection IP against an interface IP
         for iface in ret:
-            try:
-                return iface['address']
-            except Exception:
-                return None
+            addr = iface.get('address', None)
+            if addr is not None and addr == conn_addr:
+                return addr
```

Comment:
One problem is also that this is a breaking change. It might make some setups stop that worked before.

Also the documentation on `want_proxmox_nodes_ansible_host` is pretty clear: this will just give you an IP, but not necessarily the one you expect. If you want your own definition, you could use `compose` (see the last example which shows how to pick another IP).

So maybe it's best to not modify this function (or even plugin) at all?

Reply:
We can fix this just by just making sure we get an IP. And in the original code and the one I suggested above we always get the first publicly defined interface, if not available the first private one, and if not available none. Which is still in line with the original documentation.

Reply:
It's still a breaking change, if someone was using `ansible_host | default(..., true)` to replace `None` by some other value, they might suddenly get another value here. I don't use proxmox so no idea how serious this is, but generally we try to avoid breaking changes to make sure nobody's setup suddenly stops working.

Reply:
I'm not sure if the original loop is incorrect. The documentation for `want_proxmox_nodes_ansible_host` says "will use the first available interface". If you `pass` here, you are considering other interfaces than the first interface.

Reply:
Fair enough. It doesn't seem like a very useful function then, but I'll close out the PR and issue. Thanks for the input.

Reply:
It is indeed. That's why the `want_proxmox_nodes_ansible_host` option was added to disable this thing (see #2263). It's basically just there for backwards compatibility. (Maybe it's time to deprecate the default value `true` and switch it to `false` at the end of the deprecation cycle?)

Reply:
Ah, yes I wasn't aware of all of this.

It certainly wouldn't hurt to deprecate the default the value. If nothin else, it would help to make the characteristics of this function more obvious to the user. I hadn't set that option exclusively, and as such hadn't even read that it may have unexpected behavior. I just saw the erratic behavior, did a basic issue search, and immediately went into bug-fixing mode.

Apologies for the waste of time. This was my first ever PR, so I may have gotten a little overzealous.